lib/src/bump_version.dart

```diff
@@ -3,26 +3,28 @@ import 'package:sidekick_core/sidekick_core.dart';
 extension BumpVersion on Version {
   /// Returns a bumped [Version], major, minor or patch
   ///
-  /// Always bumps buildNumber when set by 1
+  /// Always strips the pre-release identifiers.
+  /// If the build information contains a single number, it is incremented.
   Version bumpVersion(VersionBumpType bumpType) {
-    final oldBuildNumber = build.firstOrNull as int?;
-    Version newVersion = this;
-    switch (bumpType) {
-      case VersionBumpType.major:
-        newVersion = nextMajor;
-        break;
-      case VersionBumpType.minor:
-        newVersion = nextMinor;
-        break;
-      case VersionBumpType.patch:
-        newVersion = nextPatch;
-        break;
-    }
+    Version newVersion = () {
+      switch (bumpType) {
+        case VersionBumpType.major:
+          return nextMajor;
+        case VersionBumpType.minor:
+          return nextMinor;
+        case VersionBumpType.patch:
+          return nextPatch;
+      }
+    }();
 
-    if (oldBuildNumber != null) {
-      final newBuildNumber = oldBuildNumber + 1;
-      // build is immutable and null if not present
-      newVersion = newVersion.copyWith(build: newBuildNumber.toString());
+    // bump of build information is only safe when it contains a single number,
+    // otherwise we don't know its incrementation schema and we leave it as is
+    if (build.whereType<int>().length == 1) {
```

Comment:
It's odd to me that [`build`](https://pub.dev/documentation/pub_semver/latest/pub_semver/Version/build.html) is `List<Object>` rather than `List<String>` but whatever 🤷

Reply:
That's because for e.g. `Version(1,2,3, build: 'foo.42.bar')`,  `build` is `['foo', 42, 'bar']` ([String, int, String])